floodsub.go

```diff
@@ -78,8 +85,10 @@ type RPC struct {
 	from peer.ID
 }
 
+type Option func(*PubSub) error
+
 // NewFloodSub returns a new FloodSub management object
-func NewFloodSub(ctx context.Context, h host.Host) *PubSub {
+func NewFloodSub(ctx context.Context, h host.Host, opts ...Option) (*PubSub, error) {
```

Comment:
I think we could avoid the interface change if we don't use any options -- do we want this for future proofing?

Reply:
Let's keep it, it's a nice interface; also we may want to pass an option for global validation throttle.